scripts/search_for_good_deck.py

```diff
@@ -0,0 +1,327 @@
+    return search_for_arena_winner_recursive(
+        decks=decks,
+        consecutive_test_wins=0,
+        consecutive_test_wins_threshold=consecutive_test_wins_threshold,
+        games_limit=games_limit_per_test,
+        p_value=p_value,
+    )
+
+
+def search_for_arena_winner_recursive(
+    decks: list[OrderedDict[str, object]],
+    consecutive_test_wins: int = 0,
+    consecutive_test_wins_threshold: int = 2,
+    games_limit: int = 100,
+    p_value: float = 0.05,
+) -> OrderedDict[str, object]:
+    print(f"\nConsecutive test wins of deck in the arena is {consecutive_test_wins}")
+
+    if consecutive_test_wins >= consecutive_test_wins_threshold:
+        return decks[0]
+
+    winner_player_index = get_index_of_best_deck(decks=copy.deepcopy(decks), games_limit=games_limit, p_value=p_value)
+
+    new_opponent_deck = create_decks(n_players=1)[0]
+    if winner_player_index is None:
+        print("No clear better deck has been identified, going to the next opponent")
+        return search_for_arena_winner_recursive(
+            decks=[copy.deepcopy(decks)[0], new_opponent_deck],
+            consecutive_test_wins=consecutive_test_wins,
+            consecutive_test_wins_threshold=consecutive_test_wins_threshold,
+            games_limit=games_limit,
+            p_value=p_value,
+        )
+
+    if winner_player_index == 1:
+        print(f"Champion has been beaten after {consecutive_test_wins} consecutive wins by")
+        print(get_deck_simple_repr(decks[1]))
+
+    return search_for_arena_winner_recursive(
+        decks=[copy.deepcopy(decks)[winner_player_index], new_opponent_deck],
+        consecutive_test_wins=consecutive_test_wins + 1 if winner_player_index == 0 else 0,
+        consecutive_test_wins_threshold=consecutive_test_wins_threshold,
+        games_limit=games_limit,
+        p_value=p_value,
+    )
+
+
+def get_index_of_best_deck(
+    decks: list[OrderedDict[str, object]], games_limit: int = 100, p_value: float = 0.05
+) -> int | None:
+    games_count = 0
+    wins_count = 0
+    while games_count < games_limit:
+        winner_player_index = simulate_one_game(decks=decks)
+        games_count += 1
+        if winner_player_index == 0:
+            wins_count += 1
+        wins_of_player_with_most_wins = max(wins_count, games_count - wins_count)
+        p_decks_are_equivalently_powerful = get_probability_of_at_least_k_heads_in_n_flips(
+            wins_of_player_with_most_wins, games_count
+        )
+        # print(
+        #    f"Probability that decks are equivalently powerful with {wins_of_player_with_most_wins} wins out of {games_count}: {p_decks_are_equivalently_powerful}"
+        # )
+        if p_decks_are_equivalently_powerful <= p_value:
+            print(f"After {games_count} games, a deck has been identified as being significantly better")
+            return 0 if wins_count / games_count >= 0.5 else 1
+
+
+def simulate_one_game(decks: list[OrderedDict[str, object]]) -> int:
```

Comment:
Maybe we could group functions like this one in a module, because otherwise we might duplicate code in different scripts (for instance, `run_competition_between_players.py` has similar code I think)